src/Microsoft.AspNetCore.Sockets.Client.Http/HttpOptions.cs

```diff
@@ -3,17 +3,29 @@
 
 using System;
 using System.Collections.Generic;
+using System.Net;
 using System.Net.Http;
 using System.Net.WebSockets;
+using System.Security.Cryptography.X509Certificates;
 
 namespace Microsoft.AspNetCore.Sockets.Client.Http
 {
     public class HttpOptions
     {
-        public HttpMessageHandler HttpMessageHandler { get; set; }
+        /// <summary>
+        /// Gets or sets a delegate for wrapping or replacing the <see cref="HttpMessageHandler"/>
+        /// that will make HTTP requests the server.
+        /// </summary>
+        public Func<HttpMessageHandler, HttpMessageHandler> HttpMessageHandler { get; set; }
```

Comment:
HttpMessageHandlerCallback? HttpMessageHandlerFactory? It's not a message handler so lets call it something else.

Reply:
Yeah, I got nervous about `HttpMessageHandlerFactory` because it's getting close to an `HttpMessageHandlerFactoryStrategyObserverBuilder` but I do think it needs to be a `Factory`.

Reply:
ConfigureMessageHandler

Reply:
Ooh, we use the `Configure` prefix for `Action<SomeOptionsType>` delegates, I do like it here.

Reply:
I like it most in the extension method name on `HubConnectionBuilder`, not sure how I feel about it on the `HttpOptions` property name. Factory might still make sense here, but we could use `.ConfigureHttpMessageHandler` as the extension method (which is what most users interact with)

Reply:
I just called it `HttpMessageHandler` on `HttpOptions` because there is just a 'WebSocketOptions' action on the same class. Should they both be prefixed with Configure?

Also it is just `WithWebSocketOptions` on the `HubConnectionBuilderHttpExtensions`, although the parameter is called `configureWebSocketOptions`.

Reply:
Let's make them both `Configure`: `ConfigureHttpMessageHandler` and `ConfigureWebSocketOptions`.